File: modules/cubex-config/src/main/java/org/cubexmc/config/NoOpMigrationStep.java
package org.cubexmc.config;

public final class NoOpMigrationStep implements MigrationStep {

    private final int fromVersion;
    private final int toVersion;
    private final String description;

    public NoOpMigrationStep(int fromVersion, int toVersion, String description) {
        this.fromVersion = fromVersion;
        this.toVersion = toVersion;
        this.description = description == null || description.isBlank() ? "No-op migration." : description;
    }

    @Override
    public int fromVersion() {
        return fromVersion;
    }

    @Override
    public int toVersion() {
        return toVersion;
    }

    @Override
    public String description() {
        return description;
    }

    @Override
    public void migrate(MigrationContext context) {
    }
}

File: modules/cubex-config/src/main/kotlin/org/cubexmc/config/NoOpMigrationStep.kt
package org.cubexmc.config

class NoOpMigrationStep(
    private val fromVersionValue: Int,
    private val toVersionValue: Int,
    description: String?,
) : MigrationStep {
    private val descriptionValue = description?.takeUnless { it.isBlank() } ?: "No-op migration."
    override fun fromVersion(): Int = fromVersionValue
    override fun toVersion(): Int = toVersionValue
    override fun description(): String = descriptionValue
    override fun migrate(context: MigrationContext) = Unit
}
